oauth_request.go

```diff
@@ -0,0 +1,53 @@
+package geddit
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+type oauthRequest struct {
+	accessToken string
+	url         string
+	useragent   string
+	values      *url.Values
+}
+
+func (r oauthRequest) getResponse() (*bytes.Buffer, error) {
+	// Determine the HTTP action.
+	var action, finalurl string
+	if r.values == nil {
+		action = "GET"
+		finalurl = r.url
+	} else {
+		action = "POST"
+		finalurl = r.url + "?" + r.values.Encode()
+	}
+
+	// Create a request and add the proper headers.
+	req, err := http.NewRequest(action, finalurl, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", r.useragent)
+	req.Header.Set("Authorization", "bearer "+r.accessToken)
+
+	// Handle the request
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
When you say "hijack" are you referring to the use of the Go `Hijack()` function, which allows the user to take over the HTTP connection? Or is your comment more of a general statement regarding the sharing of this client?